### ISSUE TITLE
Seeing unexpected token function since upgrading from 3.0.0 to 4.0.1

### ISSUE THREAD

**Issue by matt3224** [TrySound/rollup-plugin-terser#24]
Hello,

I am seeing the following with node v11.6.0, rollup v1.0.1, rollup-plugin-terser v4.0.1:
```bash
[!] (terser plugin) SyntaxError: Unexpected token function
SyntaxError: Unexpected token function

    at Object.transform (/Users/Matt/Sites/SpringBoard/app/node_modules/.registry.npmjs.org/rollup-plugin-terser/4.0.1/node_
modules/rollup-plugin-terser/transform.js:4:24)
    at execMethod (/Users/Matt/Sites/SpringBoard/app/node_modules/.registry.npmjs.org/jest-worker/23.2.0/node_modules/jest-w
orker/build/child.js:90:29)
    at process.on (/Users/Matt/Sites/SpringBoard/app/node_modules/.registry.npmjs.org/jest-worker/23.2.0/node_modules/jest-w
orker/build/child.js:40:7)
    at process.emit (events.js:188:13)
    at emit (internal/child_process.js:828:12)
    at process.internalTickCallback (internal/process/next_tick.js:72:19)
```

All my packages are up-to-date including rollup (v1.0.1). If I downgrade to rollup terser v3.0.0 the error goes away. I'm not sure what I need to do to fix the issue and would really appreciate some guidance.

Here's my rollup.config.js:
```js
// Rollup plugins
import babel from 'rollup-plugin-babel';
import resolve from 'rollup-plugin-node-resolve';
import { terser } from "rollup-plugin-terser";

export default {
    input : './scripts/application/application.js',
    output : {
        file : './scripts/application/application.min.js',
        format : 'iife',
        name : 'PROJECT'
    },
    plugins : [
        babel({
            exclude : [
                './node_modules/**',
                './scripts/parties/**',
                './scripts/polyfills/**'
            ],
            presets: [[
                '@babel/preset-env', {
                    'modules' : false,
                    'shippedProposals' : true
                }
            ]]
        }),
        resolve(),
        terser({
            sourcemap : false,
            output: {
                comments: function(node, comment) {
                    var text = comment.value;
                    var type = comment.type;
                    if (type == "comment2") {
                    // multiline comment
                        return /@preserve|@license|@cc_on/i.test(text);
                    }
                }
            }
        })
    ]
};
```

Thank you!

**Comment by TrySound** [comment]
Could you provide repository reproducing the problem?

**Comment by matt3224** [comment]
Yeah sure its here: https://github.com/matt3224/test

**Comment by matt3224** [comment]
I updated deps on the off chance 4.0.2 or rollup 1.1 fixed it but no joy

**Comment by TrySound** [comment]
Sorry for wait. Which node do you use?

**Comment by matt3224** [comment]
I use the latest, node 11.6.0

**Comment by TrySound** [comment]
Weird, I installed dependencies with npm and run `build:js` script. Everything went well.

**Comment by TrySound** [comment]
By the way I use 8.15. Could you try it too?

**Comment by matt3224** [comment]
Ok i'll try that now

**Comment by matt3224** [comment]
Right so it does work on 8.15 as you say, but not 10.15 or 11.6

**Comment by zech** [comment]
I can confirm everything works with v10.14.

Had to downgrade from 10.15.

**Comment by TrySound** [comment]
Fixed in 4.0.4. Thanks to @zech
